### PR TITLE
Remove AST redundancies and extraneous nesting.

### PR DESCRIPTION
**Pull request by jpschorr**
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-ast/src/ast.rs [hunk 1]
```diff
@@ -878,5 +768,5 @@ pub struct CustomType {
 pub struct SymbolPrimitive {
     pub value: String,
     // Optional because string literal symbols don't have case sensitivity
-    pub case: Option<CaseSensitivity>,
+    pub case: CaseSensitivity,
```
Comment (alancai98) [inline partiql-ast/src/ast.rs:771]: Above comment about optionality isn't relevant anymore

File: partiql-ast/src/ast.rs [hunk 2]
```diff
@@ -315,7 +250,6 @@ pub enum Lit {
     NationalCharStringLit(String),
     BitStringLit(String),
     HexStringLit(String),
-    DateTimeLit(DateTimeLit),
```
Comment (alancai98) [inline partiql-ast/src/ast.rs:318]: Why are `DateTimeLit` and `Param` on L348 being removed?

File: partiql-ast/src/ast.rs [hunk 3]
```diff
@@ -179,61 +157,20 @@ pub enum ConflictAction {
     DoNothing,
 }
 
-#[derive(Clone, Debug, PartialEq)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Expr {
```
Comment (alancai98) [inline partiql-ast/src/ast.rs:184]: With these ast changes, `partiql-cli` may also need updated at some point in the future (in particular [ast_to_dot.rs](https://github.com/partiql/partiql-lang-rust/blob/e2cbf2a637ee838040ec41dcd8918f585c00a938/partiql-cli/src/visualize/ast_to_dot.rs)).

File: partiql-ast/src/ast.rs (hunk 2 above)
Comment (jpschorr) [reply partiql-ast/src/ast.rs:382]: Nothing ever parsed to either of them.

Date/Time literals are parsed into `TypedLit`s at current.

File: partiql-ast/src/ast.rs (hunk 3 above)
Comment (jpschorr) [reply partiql-ast/src/ast.rs:184]: will fix

File: partiql-ast/src/ast.rs (hunk 2 above)
Comment (alancai98) [reply partiql-ast/src/ast.rs:318]: Ah, gotcha

**Review by alancai98 — APPROVED** [review]

**Pull request merged**